server/src/main/java/com/scalar/db/server/DistributedStorageService.java

```diff
@@ -206,8 +207,10 @@ public void onError(Throwable t) {
 
```

Comment:
(Sorry, it cannot comment on the code, but it's for the `onError` method)
Not really sure but is it OK not to call `responseObserver.onError(t)` when onError is called?